### DISCUSSION TITLE
What is the event triggered by expanding and contracting tree nodes?

### DISCUSSION THREAD

**Discussion by qingtingzhiyu39** [eclipse-theia/theia#14330] [General]
hi,
  My widget extends TreeWidget,i wan to know What is the event triggered by expanding and contracting tree nodes?

**Comment by msujew** [comment]
Hey @qingtingzhiyu39,

we have a `TreeExpansionService` that you can inject in your widget. See [here](https://github.com/eclipse-theia/theia/blob/7e51b89a7b5ac790399ec8ca3df4f620f69a7b77/packages/core/src/browser/tree/tree-expansion.ts#L85).
